sdk/src/test/java/com/mapbox/maps/ViewAnnotationManagerAddTest.kt

```diff
@@ -0,0 +1,228 @@
+package com.mapbox.maps
+
+import android.view.LayoutInflater
+import android.view.View
+import android.widget.FrameLayout
+import androidx.asynclayoutinflater.view.AsyncLayoutInflater
+import com.mapbox.bindgen.ExpectedFactory
+import com.mapbox.geojson.Point
+import com.mapbox.maps.viewannotation.viewAnnotationOptions
+import io.mockk.*
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+
+@RunWith(Parameterized::class)
+class ViewAnnotationManagerAddTest(
+  private val resIdActual: Int,
+  private val layoutParamsWidthActual: Int,
+  private val layoutParamsHeightActual: Int,
+  private val viewAnnotationOptionsActual: ViewAnnotationOptions,
+  private val viewAnnotationOptionsExpected: ViewAnnotationOptions,
+  private val runtimeExceptionThrown: Boolean,
+) {
+  private val mapboxMap: MapboxMap = mockk(relaxUnitFun = true)
+  private val mapView: MapView = mockk(relaxUnitFun = true)
+
+  private lateinit var viewAnnotationManager: ViewAnnotationManagerImpl
+  private lateinit var expectedView: View
+  private lateinit var frameLayoutParams: FrameLayout.LayoutParams
+
+  @Before
+  fun setUp() {
+    every { mapView.getMapboxMap() } returns mapboxMap
+    every { mapView.mapController.pluginRegistry.viewPlugins } returns mutableMapOf()
+    viewAnnotationManager = ViewAnnotationManagerImpl(mapView)
+    expectedView = mockk(relaxed = true)
+    frameLayoutParams = mockk()
+    frameLayoutParams.width = layoutParamsWidthActual
+    frameLayoutParams.height = layoutParamsHeightActual
+    every { mapView.context } returns mockk()
+    every { expectedView.layoutParams } returns frameLayoutParams
+    every { mapboxMap.addViewAnnotation(any(), any()) } returns ExpectedFactory.createNone()
+  }
+
+  @After
+  fun tearDown() {
+    every { mapboxMap.removeViewAnnotation(any()) } returns ExpectedFactory.createNone()
+    viewAnnotationManager.destroy()
+  }
+
+  @Test
+  fun addViewAnnotationWithSyncResId() {
+    mockkStatic(LayoutInflater::class)
+    every { LayoutInflater.from(any()).inflate(resIdActual, mapView, false) } returns expectedView
+    try {
```

Comment:
we need to handle both case when catching and not catching exception, tried this approach - it does not work as we are always expecting exception if using this setup while it's not true because we're using parameterized tests.